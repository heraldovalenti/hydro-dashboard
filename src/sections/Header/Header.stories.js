import React from 'react';
import Header from '.';

export default {
  title: 'Header',
  component: Header,
};

export const WithDates = () => (
  <Header startDate={'2020-03-03'} endDate={'2020-04-04'}></Header>
);

export const Loading = () => <Header />;
