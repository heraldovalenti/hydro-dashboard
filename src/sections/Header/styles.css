.header {
  position: relative;
}

.header__buttons > .header__button {
  font-size: 1rem;
}

.header__title {
  font-size: 38.4px;
  font-weight: 200;
  letter-spacing: -1.152px;
  line-height: 42.24px;
  margin: 0;
  text-transform: uppercase;
}

.header__description {
  color: rgb(70, 70, 70);

  font-size: 12px;
  margin: 3px;
}

.header__buttons {
  right: 10px;
  top: 10px;
  position: absolute;
}

.header__date {
  color: rgb(70, 70, 70);

  font-size: 14px;
  margin: 3px;
}
