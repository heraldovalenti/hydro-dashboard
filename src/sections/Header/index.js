import React, { useContext } from 'react';
import { getAmericanDateString } from '../../utils/date';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { AppDataContext } from '../../providers/AppDataProvider';
import { Tooltip, IconButton } from '@material-ui/core';

import './styles.css';

function HeaderDates({ startDate, endDate }) {
  return (
    <p className="header__date">
      {getAmericanDateString(startDate)} - {getAmericanDateString(endDate)}
    </p>
  );
}

const Header = ({
  title,
  subtitle,
  onSavePlanClick,
  onCalendarClick,
  onDiskClick,
  showDates,
}) => {
  const { fetchStartDate: startDate, fetchEndDate: endDate } = useContext(
    AppDataContext
  );
  const isDatesSet = startDate && endDate;
  let headerDates;
  if (isDatesSet) {
    headerDates = <HeaderDates startDate={startDate} endDate={endDate} />;
  } else {
    headerDates = <p className="header__date">...</p>;
  }
  return (
    <div className="header">
      <h1 className="header__title">{title}</h1>
      <div className="header__buttons">
        {onSavePlanClick && (
          <Tooltip title="Commit plan">
            <IconButton className="header__button" onClick={onSavePlanClick}>
              <FontAwesomeIcon color={'rgb(70, 70, 70)'} icon="paper-plane" />
            </IconButton>
          </Tooltip>
        )}
        {onCalendarClick && (
          <Tooltip title="Select new time period">
            <IconButton className="header__button" onClick={onCalendarClick}>
              <FontAwesomeIcon color={'rgb(70, 70, 70)'} icon="calendar-alt" />
            </IconButton>
          </Tooltip>
        )}
        {onDiskClick && (
          <Tooltip title="Save scenario">
            <IconButton className="header__button" onClick={onDiskClick}>
              <FontAwesomeIcon color={'rgb(70, 70, 70)'} icon="save" />
            </IconButton>
          </Tooltip>
        )}
      </div>
      {subtitle && <p className="header__description">{subtitle}</p>}
      {showDates && headerDates}
    </div>
  );
};

export default Header;
