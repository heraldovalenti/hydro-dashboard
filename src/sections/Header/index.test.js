import React from 'react';
import Header from './';
import renderer from 'react-test-renderer';
import StoreProvider from '../../store';

describe('<Header/>', () => {
  it('renders header with loading dates when dates were not yet passed', () => {
    const tree = renderer
      .create(
        <StoreProvider>
          <Header />
        </StoreProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders header with dates when dates loaded', () => {
    const tree = renderer
      .create(
        <StoreProvider>
          <Header startDate={'2020-03-03'} endDate={'2020-04-04'}></Header>
        </StoreProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
