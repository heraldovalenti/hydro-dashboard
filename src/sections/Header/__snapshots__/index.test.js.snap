// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders header with dates when dates loaded 1`] = `
<div
  className="header"
>
  <h1
    className="header__title"
  />
  <div
    className="header__buttons"
  />
</div>
`;

exports[`<Header/> renders header with loading dates when dates were not yet passed 1`] = `
<div
  className="header"
>
  <h1
    className="header__title"
  />
  <div
    className="header__buttons"
  />
</div>
`;
